httpbin/handlers.go

```diff
@@ -314,15 +314,26 @@ func (h *HTTPBin) Unstable(w http.ResponseWriter, r *http.Request) {
 // ResponseHeaders responds with a map of header values
 func (h *HTTPBin) ResponseHeaders(w http.ResponseWriter, r *http.Request) {
 	args := r.URL.Query()
+	body := make(map[string]interface{})
 	for k, vs := range args {
 		for _, v := range vs {
 			w.Header().Add(k, v)
 		}
+		if len(vs) == 1 {
+			body[k] = vs[0]
+		} else {
+			body[k] = vs
+		}
 	}
 	if contentType := w.Header().Get("Content-Type"); contentType == "" {
 		w.Header().Set("Content-Type", jsonContentType)
 	}
-	mustMarshalJSON(w, args)
+	body["Content-Type"] = w.Header().Get("Content-Type")
+	body["Content-Length"] = 0
+	buf := &bytes.Buffer{}
+	json.NewEncoder(buf).Encode(body)
+	body["Content-Length"] = buf.Len()
+	mustMarshalJSON(w, body)
```

Comment:
This is a bit of a strange approach, double-encoding the body just to get the content length.  Please rewrite it to do the encoding once.  The existing [`writeResponse()` helper func](https://github.com/mccutchen/go-httpbin/blob/62ec5553f2401137c24a0cc67f04c5fd278a4e4b/httpbin/helpers.go#L96-L100) might be useful here.